intervaltree/intervaltree.py

```diff
@@ -437,6 +437,17 @@ def intersection_update(self, other):
             if iv not in other:
                 self.remove(iv)
 
+    def overlap_intervals(self,other):
+        """
+        Returns a new IntervalTree consisting of intervals representing the
+        regions overlapped by at least one interval in both of self and other.
+        """
+        splits = (self | other)
+        splits.split_overlaps()
+        self_int_other =  IntervalTree(filter(lambda r: self.overlaps(r) and other.overlaps(r), splits))
+        self_int_other.merge_overlaps()
```

Comment:
Is there a reason that the overlaps have to be merged? This causes a new tree to be created a second time, which is O(nlogn).